src/modules/display.js

```diff
@@ -0,0 +1,171 @@
+    checked.type = 'checkbox';
+    checked.name = 'task-done';
+    checked.checked = false;
+    checked.id = todo.index;
+    text.type = 'text';
+    text.name = 'task';
+    text.value = todo.description;
+    text.id = todo.index;
+    text.disabled = true;
+    text.style.cursor = 'pointer';
+    text.style.background = 'none';
+    text.style.color = 'black';
+    menuIcon.innerHTML = '<i class="fa-solid fa-ellipsis-vertical" for="select"></i>';
+    deleteIcon.innerHTML = `<i class="fa-solid fa fa-trash" id="${todo.index}"></i>`;
+    deleteIcon.style.display = 'none';
+    li.id = todo.index;
+    li.appendChild(checked);
+    li.appendChild(text);
+    li.appendChild(menuIcon);
+    li.appendChild(deleteIcon);
+    list.appendChild(li);
+    if (todo.completed === true) {
+      checked.checked = true;
+      text.style.textDecoration = 'line-through';
+    }
+
+    // Drag Functions
+    function dragStart(e) {
+      dragStartId = e.target.id;
+      this.classList.add('selected');
+      setTimeout(() => this.classList.add('invisible'), 0);
+    }
+
+    function dragEnd() {
+      this.classList.remove('invisible');
+      this.classList.remove('selected');
+    }
+
+    function dragOver(e) {
+      e.preventDefault();
+    }
+
+    function dragEnter(e) {
+      e.preventDefault();
+      this.classList.add('hovered');
+    }
+
+    function dragLeave() {
+      this.classList.remove('hovered');
+    }
+
+    function dragDrop(e) {
+      let count = 1;
+      const replaceIndex = e.target.id - 1;
+      const indexNew = dragStartId - 1;
+      const [replaceValue] = data.todos.splice(indexNew, 1);
+      data.todos.splice(replaceIndex, 0, replaceValue);
+      data.todos = data.todos.map((todo) => ({
+        description: todo.description,
+        completed: todo.completed,
+        index: count++,
+      }));
+      this.classList.remove('hovered');
+      localStorage.setItem('todos', JSON.stringify(data.todos));
+      display();
+    }
+
+    // Drag event
+    li.addEventListener('dragstart', dragStart);
+    li.addEventListener('dragend', dragEnd);
+    li.addEventListener('dragover', dragOver);
+    li.addEventListener('dragenter', dragEnter);
+    li.addEventListener('dragleave', dragLeave);
+    li.addEventListener('drop', dragDrop);
+
+    const deleteTodo = (e) => {
+      const taskId = e.target.id;
+      let count = 1;
+      data.todos = data.todos.filter((todo) => todo.index !== Number(taskId));
+      data.todos = data.todos.map((todo) => {
+        if (todo.index > taskId) {
+          return {
+            description: todo.description,
+            completed: todo.completed,
+            index: count++,
+          };
+        }
+
+        return {
+          description: todo.description,
+          completed: todo.completed,
+          index: todo.index,
+        };
+      });
+      localStorage.setItem('todos', JSON.stringify(data.todos));
+      // e.target.parentElement.parentElement.remove();
```

Comment:
Inline comments of code are considered bad practice, I think your code would look more professional if you removed them. And you don't have to worry about losing those previous implementations, they will always be available in your git history

Reply:
Thank you very much for spotting this, i will make the required changes as soon as possible